src/jdk.httpserver/share/classes/sun/net/httpserver/ChunkedOutputStream.java

```diff
@@ -137,8 +137,9 @@ public void close () throws IOException {
         if (closed) {
             return;
         }
-        flush();
         try {
+            /* write any pending chunk data */
+            writeChunk();
```

Comment:
Isn't there a risk that you might write two empty chunks here if there are no pending chunks?

Reply:
yes, let me fixed that.

Reply:
Shouldn't you look at whether count > 0 as is  done in flush() ?

Reply:
corrected. good catch.